geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/FilterHelperTest.scala

```diff
@@ -65,13 +63,24 @@ class FilterHelperTest extends Specification with Mockito with LazyLogging {
   def between(dt1: DateTime, dt2: DateTime): Filter = ff.between(dtp, dt2lit(dt1), dt2lit(dt2))
   def between(dtTuple: (DateTime, DateTime)): Filter = between(dtTuple._1, dtTuple._2)
 
-  def interval(dtTuple: (DateTime, DateTime)) = new Interval(dtTuple._1, dtTuple._2)
-  def afterInterval(dt: DateTime): Interval   = new Interval(dt, max)
-  def beforeInterval(dt: DateTime): Interval  = new Interval(min, dt)
+  def interval(dtTuple: (DateTime, DateTime)) = (dtTuple._1, dtTuple._2)
+  def afterInterval(dt: DateTime): (DateTime, DateTime)  = (dt, max)
+  def beforeInterval(dt: DateTime): (DateTime, DateTime) = (min, dt)
 
-  val extractDT: (Seq[Filter]) => Interval = extractInterval(_, Some(dtFieldName))
+  val extractDT: (Seq[Filter]) => (DateTime, DateTime) = {
+    import scala.collection.JavaConversions._
+    (f) => extractIntervals(ff.and(f), dtFieldName).headOption.getOrElse(null, null)
```

Comment:
This is kind of an intermediate PR - the old code for extracting intervals returned a single interval - however, if you have an OR in your query, there might be multiple intervals that need to be returned. So to keep it simpler, currently I'm just using headOption on the intervals coming back (since we never pass an OR), but that will hopefully change in the future.